local-tests/setup/tinny-utils.ts

```diff
@@ -45,3 +45,18 @@ export function randomSolanaPrivateKey() {
   }
   return result;
 }
+
+/**
+ * Wraps a promise with a timeout.
+ * If the promise does not resolve or reject within the specified time, it will be rejected with a "Timed out" error.
+ *
+ * @param promise - The promise to wrap with a timeout.
+ * @param ms - The timeout duration in milliseconds.
+ * @returns A new promise that resolves or rejects based on the original promise or the timeout.
+ */
+export function withTimeout<T>(promise: Promise<T>, ms: number): Promise<T> {
```

Comment:
I dont think the return type is `Promise<T>` Since the race is not guaranteed to resolve `promise` and therefore might be `void` if the timeout triggers.

Reply:
the `Promise.race` method returns a promise that resolves or rejects as soon as one of the promises in the itterable resolves or rejects, means that the return type will always be `Promise<T>` if the input promise resolves or `Promise<void>` if the timeout triggers, but both are covered under the type `Promise<T>`.

Reply:
Im not sure that 'T' covers the case where the timeout promise resolves first and the value returned is void and `T` is not guaranteed to include `void` thinking the return type should look something like 
```js
withTimeout<T>(promise: Promise<T>, ms: number): Promise<T | void>
```
as to denote that the promise returned may resolve the given type specified by the promise provided, or return `void` meaning the timeout promise resolved first. Perhaps I am missing something.